tauri/src/networks/mod.rs

```diff
@@ -65,24 +64,46 @@ impl Networks {
     }
 
     pub fn get_current_network(&self) -> &Network {
+        if !self.networks.contains_key(&self.current) {
+            return self.networks.values().next().unwrap();
+        }
+
         &self.networks[&self.current]
     }
 
-    pub fn set_networks(&mut self, networks: Vec<Network>) {
-        self.networks = networks.into_iter().map(|n| (n.name.clone(), n)).collect();
-        // TODO: ensure current network still exists
-        self.save().unwrap();
+    pub fn set_networks(&mut self, networks: Vec<Network>) -> Result<()> {
+        let networks = networks.into_iter().map(|n| (n.name.clone(), n)).collect();
+
+        self.do_set_networks(networks)
     }
 
-    pub fn reset_networks(&mut self) {
-        self.networks = Network::default();
-        self.save().unwrap();
+    pub fn reset_networks(&mut self) -> Result<()> {
+        self.do_set_networks(Network::default())
     }
 
     pub fn get_current_provider(&self) -> Provider<Http> {
         self.get_current_network().get_provider()
     }
 
+    fn broadcast_network_changed(&self) -> Result<()> {
+        self.notify_peers();
+        self.window_snd.send(app::Notify::NetworkChanged.into())?;
+
+        Ok(())
+    }
+
+    fn do_set_networks(&mut self, networks: HashMap<String, Network>) -> Result<()> {
+        self.networks = networks;
+
+        if !self.networks.contains_key(&self.current) {
+            let name = self.networks.values().next().unwrap().name.clone();
```

Comment:
this is not deterministic. HashMaps don't have a fixed order, as they would in JS, so this essentially means the current network will be a random one, and may not even be the same between runs, or between computers.

That's also one of the reasons why the `set_networks` state is a `Vec` and not a `HashMap` as well.

my suggestion: have this function receive a `Vec<Network>` instead (which will also simplify the `set_networks` call), and change `reset_networks` to create a vec with a hardcoded order`

then you can use `networks[0].name` as the fallback one

Reply:
> so this essentially means the current network will be a random one, and may not even be the same between runs, or between computers.

I need some clarification. This code is only executed when you update the list of networks and the new list doesn't include the currently selected network. Sure, it picks a random one, but that one will be persisted, and the subsequent runs will always use that until the user changes it. Does it matter that the selection is random? I'm happy to change it, but it feels a bit irrelevant. But feel free just to say that you think it's important; there's no need to make it a big thing.